src/NHibernate.Test/TestCase.cs

```diff
@@ -292,7 +292,12 @@ protected virtual void DropSchema()
 
 		protected virtual DebugSessionFactory BuildSessionFactory()
 		{
-			return new DebugSessionFactory(cfg.BuildSessionFactory());
+			var factory = new DebugSessionFactory(cfg.BuildSessionFactory());
+			// Avoid ORA-12704 in tests, due to NHibernate defaulting String to NVarchar2 when creating tables while
+			// ODP.Net defaults it to Varchar2 (which does support Unicode too, depending on server) for parameters.
+			if (factory.ConnectionProvider?.Driver is OracleManagedDataClientDriver oracleManagedDriver)
+				oracleManagedDriver.UseNVarchar2ForStringParameter = true;
```

Comment:
I have added that setting on the oracle managed driver. There are ore detail on the parameter comments. The culprit is what I see as a mismatch between NHibernate Oracle Dialect, matching String to `NVarchar2`, while ODP matches it to `Varchar2` instead (which can be Unicode too, unlike with SQL Server). Oracle refuses to mix those string types for a number of operations, causing issues in some queries, because NHibernate creates columns as `NVarchar2` while string parameters are transmitted as `Varchar2`.

Maybe we should change that in Oracle Dialect for re-aligning on ODP defaults. But that would then requires another Jira I think.